src/vmm/src/devices/virtio/block/virtio/request.rs

```diff
@@ -313,6 +334,52 @@ impl Request {
                     return Err(VirtioBlockError::InvalidDataLength);
                 }
             }
+            RequestType::Discard => {
+                // Get data descriptor
+                let data_desc = avail_desc.next_descriptor()
+                    .ok_or(VirtioBlockError::DescriptorChainTooShort)?;
```

Comment:
This is not a place to parse the request. It needs to be done above. There is already a place where `data_desc` is assigned.